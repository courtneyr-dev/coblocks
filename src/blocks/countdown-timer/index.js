/**
 * External dependencies
 */
import { TimerIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies.
 */
import edit from './edit';
// import example from './example';
import metadata from './block.json';
import save from './save';

/**
 * WordPress dependencies.
 */
import { Icon } from '@wordpress/components';
import { __, _x } from '@wordpress/i18n';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	description: __( 'Add a countdown timer to promote limited time offers or upcoming events.', 'coblocks' ),
	edit,
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'countdown', 'coblocks' ),
		/* translators: block keyword */
		__( 'timer', 'coblocks' ),
	],
	save,
	supports: {
		align: [ 'wide', 'full' ],
		multiple: false,
	},
	title: _x( 'Countdown Timer', 'block name', 'coblocks' ),
};

export { metadata, name, category, settings };
