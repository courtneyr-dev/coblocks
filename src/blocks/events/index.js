/**
 * External dependencies
 */
import { EventsIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies.
 */
import edit from './edit';
import example from './example';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';
import deprecated from './deprecated';

/**
 * WordPress dependencies.
 */
import { __, _x } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Block constants.
 */
const { name, category } = metadata;

const settings = {
	title: _x( 'Events', 'block name', 'coblocks' ),
	description: __( 'Add a list of events or display events from a public calendar.', 'coblocks' ),
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'calendar', 'coblocks' ),
		/* translators: block keyword */
		__( 'date', 'coblocks' ),
	],
	supports: {
		align: [ 'wide', 'full' ],
	},
	// example,
	transforms,
	edit,
	save,
	deprecated
};

export { metadata, name, category, settings };
