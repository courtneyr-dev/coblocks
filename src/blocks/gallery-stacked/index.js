/**
 * External dependencies
 */
import { GalleryStackedIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import { BLOCK_VARIATION_GALLERY_STACKED } from '../../block-variations/core/gallery';
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { dispatch } from '@wordpress/data';
import { createBlock, switchToBlockType } from '@wordpress/blocks';

/**
 * Block constants
 */
const { name, category } = metadata;

const settings = {
	attributes: {
		ids: { type: 'array' },
		images: { type: 'array' },
		// linkTo: { type: 'string' },
		// sizeSlug: { type: 'string' },
	},
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );
		replaceBlocks(
			[ props.clientId ],
			switchToBlockType( props, 'core/gallery' )
		);
		return null;
	},
	parent: [],
	save: () => null,
	transforms: {
		to: [
			{
				blocks: [ 'core/gallery' ],
				transform: ( attributes ) => {
					return createBlock(
						'core/gallery',
						Object.assign( {}, attributes, BLOCK_VARIATION_GALLERY_STACKED.attributes )
					);
				},
				type: 'block',
			},
		],
	},
};

export { name, category, icon, metadata, settings };
