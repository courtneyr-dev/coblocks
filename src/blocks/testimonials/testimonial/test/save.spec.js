/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';
import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

describe( 'coblocks/testimonial', () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render empty when Name is missing', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with content', () => {
		block.attributes.name = 'Some name';
		block.attributes.role = 'Some role';
		block.attributes.text = 'Some text';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Some name' );
		expect( serializedBlock ).toContain( 'Some role' );
		expect( serializedBlock ).toContain( 'Some text' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with the "conversation" style', () => {
		block.attributes.name = 'Some name';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Some name' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with the "horizontal" style', () => {
		block.attributes.name = 'Some name';
		block.attributes.styleName = 'horizontal';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'Some name' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with image', () => {
		block.attributes.name = 'Some name';
		block.attributes.url = 'http://some.url/someimage.jpg';
		block.attributes.showImage = true;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'http://some.url/someimage.jpg' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with focal point on image', () => {
		block.attributes.name = 'Some name';
		block.attributes.url = 'http://some.url/someimage.jpg';
		block.attributes.focalPoint = { x: 0.5, y: 1 };
		block.attributes.showImage = true;
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'background-position:50% 100%' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with background color', () => {
		block.attributes.name = 'Some name';
		block.attributes.backgroundColor = 'primary';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"backgroundColor":"primary"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with custom background color', () => {
		block.attributes.name = 'Some name';
		block.attributes.customBackgroundColor = '#745151';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"customBackgroundColor":"#745151"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with text color', () => {
		block.attributes.name = 'Some name';
		block.attributes.textColor = 'background';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"textColor":"background"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with custom text color', () => {
		block.attributes.name = 'Some name';
		block.attributes.customTextColor = '#ffda7c';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '{"customTextColor":"#ffda7c"}' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with bubble background color', () => {
		block.attributes.name = 'Some name';
		block.attributes.bubbleBackgroundColor = 'primary';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'has-primary-background-color' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with custom bubble background color', () => {
		block.attributes.name = 'Some name';
		block.attributes.customBubbleBackgroundColor = '#745151';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"customBubbleBackgroundColor":"#745151"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with bubble text color', () => {
		block.attributes.name = 'Some name';
		block.attributes.bubbleTextColor = 'background';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"bubbleTextColor":"background"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should render with custom bubble text color', () => {
		block.attributes.name = 'Some name';
		block.attributes.customBubbleTextColor = '#ffda7c';
		block.attributes.styleName = 'conversation';
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( '"customBubbleTextColor":"#ffda7c"' );
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );
