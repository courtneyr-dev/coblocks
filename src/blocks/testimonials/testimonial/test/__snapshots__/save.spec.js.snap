// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/testimonial should render empty when Name is missing 1`] = `"<!-- wp:coblocks/testimonial /-->"`;

exports[`coblocks/testimonial should render with background color 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\",\\"backgroundColor\\":\\"primary\\"} -->
<div class=\\"wp-block-coblocks-testimonial has-primary-background-color\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front has-primary-background-color\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with bubble background color 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\",\\"bubbleBackgroundColor\\":\\"primary\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text has-primary-background-color\\" itemprop=\\"description\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back has-primary-background-color\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with bubble text color 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\",\\"bubbleTextColor\\":\\"background\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text has-background-color\\" itemprop=\\"description\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with content 1`] = `
"<!-- wp:coblocks/testimonial -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4><p class=\\"wp-block-coblocks-testimonial__heading__role\\">Some role</p></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\">Some text</p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with custom background color 1`] = `
"<!-- wp:coblocks/testimonial {\\"customBackgroundColor\\":\\"#745151\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\" style=\\"background-color:#745151;background:#745151\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with custom bubble background color 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\",\\"customBubbleBackgroundColor\\":\\"#745151\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\" style=\\"background-color:#745151\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back\\" style=\\"background-color:#745151\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with custom bubble text color 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\",\\"customBubbleTextColor\\":\\"#ffda7c\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\" style=\\"color:#ffda7c\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with custom text color 1`] = `
"<!-- wp:coblocks/testimonial {\\"customTextColor\\":\\"#ffda7c\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\" style=\\"color:#ffda7c\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with focal point on image 1`] = `
"<!-- wp:coblocks/testimonial {\\"focalPoint\\":{\\"x\\":0.5,\\"y\\":1},\\"showImage\\":true} -->
<div class=\\"wp-block-coblocks-testimonial\\" style=\\"background-position:50% 100%\\"><figure class=\\"wp-block-coblocks-testimonial__image\\" style=\\"height:100px;width:100px\\"><img alt=\\"\\" src=\\"http://some.url/someimage.jpg\\" style=\\"object-position:50% 100%\\"/></figure><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with image 1`] = `
"<!-- wp:coblocks/testimonial {\\"showImage\\":true} -->
<div class=\\"wp-block-coblocks-testimonial\\"><figure class=\\"wp-block-coblocks-testimonial__image\\" style=\\"height:100px;width:100px\\"><img alt=\\"\\" src=\\"http://some.url/someimage.jpg\\"/></figure><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with text color 1`] = `
"<!-- wp:coblocks/testimonial {\\"textColor\\":\\"background\\"} -->
<div class=\\"wp-block-coblocks-testimonial has-background-color\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with the "conversation" style 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"conversation\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__text-bubble\\"><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-back\\"></span><span class=\\"wp-block-coblocks-testimonial__text-bubble__tip-front\\"></span></div><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div></div></div>
<!-- /wp:coblocks/testimonial -->"
`;

exports[`coblocks/testimonial should render with the "horizontal" style 1`] = `
"<!-- wp:coblocks/testimonial {\\"styleName\\":\\"horizontal\\"} -->
<div class=\\"wp-block-coblocks-testimonial\\"><div class=\\"wp-block-coblocks-testimonial__content\\"><div class=\\"wp-block-coblocks-testimonial__heading\\"><h4 class=\\"wp-block-coblocks-testimonial__heading__name\\">Some name</h4></div><p class=\\"wp-block-coblocks-testimonial__text\\" itemprop=\\"description\\"></p></div></div>
<!-- /wp:coblocks/testimonial -->"
`;
